### ISSUE TITLE
33 - System.IO.InvalidDataException: 'Found invalid data while decoding.'

### ISSUE THREAD

**Issue by KMastalerz** [MarkPflug/Sylvan.Data.Excel#34]
Yet another one, sorry you must hate me at this point. 

**Crash at create:**

```
                var options = new SylExcel.ExcelDataReaderOptions { Schema = SylExcel.ExcelSchema.Default, GetErrorAsNull = true };
                //collect data
                using (SylExcel.ExcelDataReader excelDataReader = SylExcel.ExcelDataReader.Create(Setup.FilePath, options))
```

**Exception:** 

System.IO.InvalidDataException: 'Found invalid data while decoding.'

**File**
[Copy of dmsg-11-50-a3-a.xls](https://github.com/MarkPflug/Sylvan.Data.Excel/files/8108383/Copy.of.dmsg-11-50-a3-a.xls)

Additionaly, it has error with mapping class on ExRate due to Venezuelan "NA", I fixed that with ExRateStr but just to let you know :)

```
    public class ExRateItem
    {
        //fixer
        private string _ExRateStr;
        public string ExRateStr
        {
            get { return _ExRateStr; }
            set
            {
                _ExRateStr = value;
                try
                {
                    ExRate = decimal.Parse(ExRateStr);
                }
                catch
                {
                    ExRate = 0;
                }
            }
        }

        public string Country { get; set; }
        public string CurrencyName { get; set; }
        public string BaseCurrency { get; set; }
        public string LocalCurrency { get; set; }
        public int Year { get; set; }
        public int Month { get; set; }
        public decimal? ExRate { get; set; }
        public bool isActive { get; set; }
        public string AddedBy { get; set; }
        public DateTime? AddedDate { get; set; }
        public string ModifiedBy { get; set; }
        public DateTime? ModifiedDate { get; set; }
    }
```

**Comment by KMastalerz** [comment]
And another one, at same stage with exception:  "System.IO.InvalidDataException: 'Found invalid data while decoding.'"
[Infinity Codes_20220218153829.xlsx](https://github.com/MarkPflug/Sylvan.Data.Excel/files/8108524/Infinity.Codes_20220218153829.xlsx)

**Comment by KMastalerz** [comment]
And another one, at same stage with exception: "System.IO.InvalidDataException: 'Found invalid data while decoding.'"

It's sent via email.

**Comment by KMastalerz** [comment]
Same error occurs on all files that previously had been working, this is something with new release

**Comment by MarkPflug** [comment]
All files? All .xls files?

**Comment by MarkPflug** [comment]
The .xlsx issue I have a fix for. This .xls file, however, looks very strange to me. Do you know where these .xls files are coming from? Are they actually created in Excel?

**Comment by KMastalerz** [comment]
The other one was also .xlsx. 

Im not sure of how the .xls is populated i source it from sharepoint (it's saved before sylvan is used). But this one works, when i take data as string and parse it to correct property.

**Comment by KMastalerz** [comment]
Let me know when fix for .xlsx will hit, ill test all the files i have against it again tomorrow :)

**Comment by MarkPflug** [comment]
> Yet another one, sorry you must hate me at this point.

Absolutely not. You are dealing with bugs in *my* code, so you should be the one who has regrets about choosing my library. Your example files are extremely helpful, so thanks for that.

I just published 0.1.9 to nuget which has fixes for these files.
